.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+default_language_version:
+  python: python3
+
+exclude: ^$|instrument
+
+repos:
+
+  # Run fast code improvement/checks before running PR specific helpers.
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: 2618198e9658cccb4a53f04af0f7d642109f3b54
+    hooks:
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+      - id: check-added-large-files
+        args: ['--maxkb=4096']
+      - id: check-xml
+      - id: check-yaml
+
+  - repo: https://github.com/mantidproject/pre-commit-hooks
```

Comment:
clang-format and yapf should be done in separate PRs as they have knock-on effects for changing lots of other files. Even not on the build server, developers will be changing a ton of files unnecessarily.

Reply:
Otherwise, I'm happy with what is proposed here. I am just concerned that the "normal" developer will make a bunch of conflicts.

Reply:
Yep. Totally agree. We were going to wait until the last minute to open that PR before we would then merge this.